### PR TITLE
Feature/us7090 improve my groups speed

### PR DESCRIPTION
**Pull request by JoeKer1**

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation.Test/Services/GroupRepositoryTest.cs [hunk 1]
```diff
@@ -497,6 +553,16 @@ public void GetMyGroupsByType()
         }
 
         [Test]
+        public void GetMyGroupsByTypeRest()
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation.Test/Services/GroupRepositoryTest.cs]: As Clara Peller once said, "Where's the test?"

File: Gateway/MinistryPlatform.Translation/MinistryPlatform.Translation.csproj [hunk 2]
```diff
@@ -124,9 +124,11 @@
   <ItemGroup>
     <Compile Include="Exceptions\DonationNotFoundException.cs" />
     <Compile Include="Exceptions\PaymentNotFoundException.cs" />
-    <Compile Include="Models\Finder\SpPinDto.cs" />
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation/MinistryPlatform.Translation.csproj]: Possible merge conflict?  Why was this removed?

File: Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs [hunk 3]
```diff
@@ -520,52 +520,18 @@ private int GetSearchPageViewId(int groupTypeId)
...
 
             if (activeGroups)
-            {
-                participants = ministryPlatformService.GetSubpageViewRecords(GroupsParticipantsSubPageId, groupId, token);
-            }
-            else
-            {
-                participants = ministryPlatformService.GetSubpageViewRecords(CurrentGroupsParticipantsSubPage, groupId, token);
-            }
+                filter += " AND Group_ID_TABLE.Start_Date <= GETDATE() AND (Group_ID_Table.End_Date IS NULL OR Group_ID_TABLE.End_Date >= GETDATE())";
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs]: I think one of our "standards" says to enclose any block in curly braces, even if only one line.

File: Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs [hunk 4]
```diff
@@ -822,8 +788,33 @@ public List<MpGroup> GetGroupsByTypeForParticipant(string token, int participant
...
+            {
+                filter += $" AND Group_ID_Table.Group_ID = {groupId}";
+            }
+
+            var groupDetails = _ministryPlatformRestRepository.UsingAuthenticationToken(token).Search<MpGroupParticipant, MpGroup>(filter, columns);
+
+            return groupDetails;
+        }
+
+        [Obsolete("This call has been replaced by GetMyGroupsByType() which uses the rest API.")]
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs]: Nice use of obsolete!

File: Gateway/MinistryPlatform.Translation/Repositories/ObjectAttributeRepository.cs [hunk 5]
```diff
@@ -19,37 +20,30 @@ namespace MinistryPlatform.Translation.Repositories
...
         private readonly IMinistryPlatformService _ministryPlatformService;
+        private readonly IMinistryPlatformRestRepository _ministryPlatformRest;
 
         private readonly ILog _logger = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
 
         public ObjectAttributeRepository(IAuthenticationRepository authenticationService,
             IConfigurationWrapper configurationWrapper,
-            IMinistryPlatformService ministryPlatformService)
+            IMinistryPlatformService ministryPlatformService,
+            IMinistryPlatformRestRepository ministryPlatformRest)
```
Comment (IngageKrizJ) [inline Gateway/MinistryPlatform.Translation/Repositories/ObjectAttributeRepository.cs]: Might want to verify that AsyncJobs still runs locally...

File: Gateway/MinistryPlatform.Translation.Test/Services/GroupRepositoryTest.cs (hunk 1 above)
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation.Test/Services/GroupRepositoryTest.cs]: hah, the test is actually called TestGetGroupsForParticipantByTypeOrID and I forgot to remove this test.

File: Gateway/MinistryPlatform.Translation/MinistryPlatform.Translation.csproj (hunk 2 above)
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation/MinistryPlatform.Translation.csproj]: yeah that was a merge conflict I put it back

File: Gateway/MinistryPlatform.Translation.Test/Services/GroupRepositoryTest.cs (hunk 1 above)
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation.Test/Services/GroupRepositoryTest.cs]: Also Who is Clara Peller?

File: Gateway/MinistryPlatform.Translation/Repositories/ObjectAttributeRepository.cs (hunk 5 above)
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation/Repositories/ObjectAttributeRepository.cs]: Holy moly it does still work

File: Gateway/MinistryPlatform.Translation/Models/MpGroup.cs [hunk 6]
```diff
@@ -1,59 +1,134 @@
...
+using Newtonsoft.Json.Linq;
 
 
-namespace MinistryPlatform.Translation.Models
+namespace MinistryPlatform.Translation.Models 
 {
-[MpRestApiTable(Name = "Groups")]
-    public class MpGroup
+    [MpRestApiTable(Name = "Groups")]
+    public class MpGroup : MpBaseDto
```
Comment (charlie-retzler) [inline Gateway/MinistryPlatform.Translation/Models/MpGroup.cs]: Do we pass MpGroup to JavaScript client at all? If so JsonProperties could be different than what they were.

File: Gateway/MinistryPlatform.Translation/Models/MpObjectAttribute.cs [hunk 7]
```diff
@@ -1,20 +1,38 @@
 using System;
+using Crossroads.Web.Common.MinistryPlatform;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
 
 namespace MinistryPlatform.Translation.Models
 {
+    [MpRestApiTable(Name = "Attribute_Categories")]
     public class MpObjectAttribute
```
Comment (charlie-retzler) [inline Gateway/MinistryPlatform.Translation/Models/MpObjectAttribute.cs]: same thing here. Does this ever get leaked to the front end? I hope not, but should check just in case.

File: Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs (hunk 4 above)
Comment (charlie-retzler) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs]: I'm still going to ask, is this used anymore? If so we should get rid of it entirely.

File: Gateway/MinistryPlatform.Translation/Repositories/ObjectAttributeRepository.cs [hunk 8]
```diff
@@ -19,37 +20,30 @@ namespace MinistryPlatform.Translation.Repositories
...
-
-                AttributeId = record.ToInt("Attribute_ID"),
-                StartDate = record.ToDate("Start_Date"),
-                EndDate = record.ToNullableDate("End_Date"),
-                Notes = record.ToString("Notes"),
-                AttributeTypeId = record.ToInt("Attribute_Type_ID"),
-                AttributeTypeName = record.ToString("Attribute_Type")
-            }).ToList();
+            if (attributeTypeIdFilter != null)
+                filter += $"AND Attribute_ID_Table.Attribute_Type_ID = {attributeTypeIdFilter}";
```
Comment (charlie-retzler) [inline Gateway/MinistryPlatform.Translation/Repositories/ObjectAttributeRepository.cs]: Should there be a space at start of this line, or end of other filter?

File: Gateway/crds-angular/Services/ObjectAttributeService.cs [hunk 9]
```diff
@@ -40,7 +40,7 @@ public ObjectAllAttributesDTO GetObjectAttributes(string token, int objectId, Mp
 
         public ObjectAllAttributesDTO GetObjectAttributes(string token, int objectId, MpObjectAttributeConfiguration configuration, List<MpAttribute> mpAttributes)
         {
-            var mpObjectAttributes = _mpObjectAttributeService.GetCurrentObjectAttributes(token, objectId, configuration);
+            var mpObjectAttributes = _mpObjectAttributeService.GetCurrentObjectAttributes(_apiUserService.GetToken(), objectId, configuration);
```
Comment (charlie-retzler) [inline Gateway/crds-angular/Services/ObjectAttributeService.cs]: Why we get new token here when we have one passed in?

File: Gateway/MinistryPlatform.Translation/Models/MpGroup.cs (hunk 6 above)
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation/Models/MpGroup.cs]: Nah, we pass GroupDTO to javascript client I believe.  One exception may be Group search, but it seems to work (?)

File: Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs (hunk 4 above)
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupRepository.cs]: it is

File: Gateway/crds-angular/Services/ObjectAttributeService.cs (hunk 9 above)
Comment (JoeKer1) [inline Gateway/crds-angular/Services/ObjectAttributeService.cs]: Normal users didn't have access to the tables. I could've given the platform user access but i didn't think that was great.

Comment (charlie-retzler) [inline Gateway/crds-angular/Services/ObjectAttributeService.cs]: Per Andy - It's because the end user doesn't have permissions to the Contact_Attributes, Group_Attributes pages directly. For the old API we did have them through the Contact / Group pages Attributes.
